File: zcrash_lib/src/main/java/com/android/car/zcrash_lib/demo/NativeLib.java
package com.android.car.zcrash_lib.demo;

public class NativeLib {

    public NativeLib() {
        System.loadLibrary("zcrash");
    }

    public static String staticName= "Static Cat";
    public String name = "Cat";

    public native String stringFromJNI();
    public native void visitField();
    public native Person createPerson();
    public native Person[] createPersons();
    public native Person[] getPersons(String[] names);
}

File: zcrash_lib/src/main/java/com/android/car/zcrash_lib/NativeLib.kt
package com.android.car.zcrash_lib

class NativeLib {

    private val name = "Cat"
    /**
     * A native method that is implemented by the 'zcrash_lib' native library,
     * which is packaged with this application.
     */
    external fun stringFromJNI(): String
    external fun visitField()

    companion object {
        // Used to load the 'zcrash_lib' library on application startup.
        init {
            System.loadLibrary("zcrash_lib")
        }
    }
}